src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs

```diff
@@ -170,36 +168,29 @@ public void ItFailsToSetGUISubsystemWithWrongDefault()
         [PlatformSpecific(TestPlatforms.AnyUnix)]
         public void ItGeneratesExecutableImage()
         {
-            using (TestDirectory testDirectory = TestDirectory.Create())
-            {
-                string sourceAppHostMock = PrepareAppHostMockFile(testDirectory);
-                string destinationFilePath = Path.Combine(testDirectory.Path, "DestinationAppHost.exe.mock");
-                string appBinaryFilePath = "Test/App/Binary/Path.dll";
+            using TestDirectory testDirectory = TestDirectory.Create();
+            string sourceAppHostMock = PrepareAppHostMockFile(testDirectory);
+            string destinationFilePath = Path.Combine(testDirectory.Path, "DestinationAppHost.exe.mock");
+            string appBinaryFilePath = "Test/App/Binary/Path.dll";
 
-                chmod(sourceAppHostMock, Convert.ToInt32("755", 8)) // match installed permissions: -rwxr-xr-x
-                    .Should()
-                    .NotBe(-1);
+            const UnixFileMode ExecutablePermissions = UnixFileMode.UserRead | UnixFileMode.UserExecute | UnixFileMode.UserWrite |
+                UnixFileMode.GroupRead | UnixFileMode.GroupExecute |
+                UnixFileMode.OtherRead | UnixFileMode.OtherExecute;
 
-                GetLastError()
-                    .Should()
-                    .NotBe(4); // EINTR
-
-                GetFilePermissionValue(sourceAppHostMock)
-                    .Should()
-                    .Be(Convert.ToInt32("755", 8));
-
-                HostWriter.CreateAppHost(
-                    sourceAppHostMock,
-                    destinationFilePath,
-                    appBinaryFilePath,
-                    windowsGraphicalUserInterface: true);
+            File.SetUnixFileMode(sourceAppHostMock, ExecutablePermissions);
```

Comment:
The conversion looks fine, but I'm curious what this test is actually testing.  I assumed from the name of the test that it was meant to test in part whether the permissions there by default enabled execution, but this is explicitly setting permissions to make it executable.

Reply:
Agreed, we can change it to have no permissions (or different permissions) before the `CreateAppHost` call. I will update.

The original issue was https://github.com/dotnet/runtime/issues/3669 (redirected from dotnet/core-setup repo). The apphost template binary packaged with the sdk had incorrect permissions, so HostModel was fixed to always set the permissions of application to 755 after creation. This test is asserting "created app has executable permission regardless of permission on apphost template binary".